### ISSUE TITLE
Imperfect Playback of annotated region.

### ISSUE THREAD

**Issue by Sean1572** [Sean1572/audino#4]
Annotated region will play but stop playing slightly before the end of the region. It was noted the playback would stop at certain places as if the playback got stuck on a particular section of the region.

Steps to fix:
Get fresh clone of audio and see if this issue appears there

**Comment by Sean1572** [comment]
This seems to occur in clips in which the scroll becomes slightly ahead of the progress bar
![image](https://user-images.githubusercontent.com/25123131/103032897-03c87d00-4516-11eb-8543-0daab9e17e4f.png)
In this example, the bar is not centered as most clips are when using the scroll wheel

**Comment by Sean1572** [comment]
I'm going through and labeling all clips in set 1 with this issue for testing

**Comment by Sean1572** [comment]
So far it looks like the issue is not with long clips but clips that are about a min long to 2 min long (within that range)
Perhaps look into how the progress bar is synced with the scroll wheel?

**Comment by Sean1572** [comment]
On these clips, clicking the image does not center the progress bar directly on the cursor

**Comment by Sean1572** [comment]
Making the window smaller fixes the problem, the issue doesn't seem to be the progress bar but rather the size of the window used for the spectrogram

**Comment by Sean1572** [comment]
Also important to note: this issue is not being caused by the spectrogram plugin, it seems to be an issue of the scroll bar deciding to start scrolling too early/too fast

**Comment by Sean1572** [comment]
Tested turning off autocentering, it looks like its a syncing issue with audio and spectrograph and the offset during autocentering is a symptom of that not the cause.

**Comment by Sean1572** [comment]
UPDATE: I've been writing my updates in the wrong section. I'm going to move them over in a sec

**Comment by Sean1572** [comment]
One issue that I thought this was being caused by was having the system render serval small canvas instead of one big one.
However, increasing the max canvas size has not fixed this issue (see #5 ). Thus, this could be a bug within annotate.js on region playback or the region plugin itself

**Comment by Sean1572** [comment]
regions.js in the plugin's folder of wave surfer rounds the start and end times before deciding when to fire the event for playback reaching the end of the region. I'm going to try to make this more sensitive by multiplying all the times by 100 instead of 10 which was being done natively

**Comment by Sean1572** [comment]
Commenting out the code which fire the stop once the code is complete in annotate.js seems to at least play the last bit of audio while the progress bar appears to jump to the end? (although the clip is being played). 

I'm going to try one more solution (telling event listeners that the the clip isn't done manually in annotate.js
```
if (wavesurfer.getCurrentTime() < r.end) {
          console.log("fail")
          //below is done automatically in regions.js if the audio is less then the end and greater than start
          //this code techically shouldn't even be able to run if region.js does it job but i'm going to see what happens
          r.firedIn = true;
          r.firedOut = false;
          r.fireEvent('in');
          r.wavesurfer.fireEvent('region-in', r);
        } else {
          wavesurfer.pause();
        }
```

**Comment by Sean1572** [comment]
Even for clips that seem to be fine, the playback ends early by .1 seconds. The solution presented above doesn't actually even run wave.surfer.pause() if it fails (it should call again at least once).

**Comment by Sean1572** [comment]
I realized I was still using the npm installation of audino instead of the local one I made. I connected it and it seems as though the issue is fixed? A new bug has popped up with this where the audio will not play right away when selecting a  region from another region. I think this is caused by the changes I just made so I'm referencing the project #11.

updates on that issue will move to the other issue log. Audio seems to be playing on playback well.
